sway/desktop/output.c

```diff
@@ -330,9 +382,12 @@ static void handle_frame(struct wl_listener *listener, void *user_data) {
 
 	int delay = msec_until_refresh - output->max_render_time;
 
+	struct sway_view *fullscreen_view = output_get_fullscreen_view(output);
+	output->enable_tearing = output_can_tear_fullscreen_view(output, fullscreen_view);
+
 	// If the delay is less than 1 millisecond (which is the least we can wait)
-	// then just render right away.
-	if (delay < 1) {
+	// or if the output is allowed to tear, then just render right away.
+	if (delay < 1 || output->enable_tearing) {
```

Comment:
Same here: not sure we want to mess with output timings here. The default settings should result in `delay == 0`.

Reply:
Unfortunately, changing this DID make a difference. When i removed the check for tearing enabled, and ran `vkcube --present_mode 0` in fullscreen, the cube looked like the upper half was vsync-ed, and the bottom half was tearing. Like a mix of both.

If this is unexpected, then maybe we have a bug somewhere else?

Reply:
Can you print `msec_until_refresh` and `output->max_render_time`? Both should be zero with a default config.

Reply:
Ah yes i did have `max_render_time` set in my config, oops. It does appear to work correctly with `max_render_time off`. 

The question is whether we want to ignore the setting for `max_render_time` when tearing is requested? As tearing would imply that we want to render as fast as possible?

Reply:
Removed the extra check.

Perhaps we could mention in the doc that `max_render_time` should not be set for tearing to work correctly?

Reply:
Hm, to me tearing works "correctly" when `max_render_time` is set, because I see these as completely orthogonal. For instance, when setting `max_render_time` to a very low value and combining it with tearing, one can render closer to the vsync and make it so in case rendering takes too long the frame tears with the old frame for the top and the new frame for the bottom.

+1 for specifying in the docs that disabling `max_render_time` is the behavior most people will want, by flipping as soon as a client updates.

Reply:
Ah, yes I didn't think about that. This is what's called "FIFO relaxed" behavior, right?

Reply:
It's a bit similar: if we're on time it shouldn't tear, if we're late it should tear. However, there is no logic to force vsync when we're on time, and it's not a FIFO because there's no queue in which client frames accumulate. So still a bit different.

Reply:
Added an extra note about the `max_render_time` setting under the `allow_tearing` option in the `sway-output` man page.